### PR TITLE
add IBCClientHeader func for endpoint and update tests

### PR DESCRIPTION
**Pull request by expertdicer**
## Description

closes: #4003

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by colin-axner** [comment]
I can push the changes next week if you don't get a chance before then

File: testing/chain.go [hunk 1]
```diff
@@ -663,3 +625,27 @@ func (chain *TestChain) DeleteKey(key []byte) {
 	kvStore := chain.GetContext().KVStore(storeKey)
 	kvStore.Delete(key)
 }
+
+// IBCClientHeader will construct a 07-tendermint Header to update the light client
+// on the counterparty chain. The trustedHeight must be passed in as a non-zero height.
+func (chain *TestChain) IBCClientHeader(header *ibctm.Header, trustedHeight clienttypes.Height) (*ibctm.Header, error) {
```
Comment (colin-axner) [inline testing/chain.go:631]: given that we don't currently store historical headers (we only keep the latest committed header), we could drop the header arg and assume that. But fine as is, might be more flexible for future testing scenarios

**Review by colin-axner — APPROVED** [review]
Nice!

**Comment by colin-axner** [comment]
Thanks @crodriguezvega for applying the changes!

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**